### PR TITLE
Allow configuring the Spanner API endpoint used by HarbourBridge

### PR DESCRIPTION
**Pull request by aseering**
(Per offline discussion with the HarbourBridge team)

### CODE REVIEW

File: conversion/conversion.go
Comment (agasheesh) [inline conversion/conversion.go:841]: Consider moving NewDatabaseAdminClientWithEndpoint (or NewDatabaseAdminClient, please see comment above) here and renaming GetClient to NewDatabaseClientWithEndPoint (or NewDatabaseClient correspondingly) for consistency.

File: conversion/conversion.go
Comment (agasheesh) [inline conversion/conversion.go:426]: We don't need to have WithEndpoint suffix in the function name since we don't plan to have two separate functions in any case (one "WithEndpoint", other without).

**Review by agasheesh — APPROVED** [review]
Thanks for contributing this change :)

File: conversion/conversion.go
Comment (agasheesh) [inline conversion/conversion.go:421]: We need to create a similar client for InstanceAdmin as well.

File: conversion/conversion.go
Comment (aseering) [reply conversion/conversion.go]: Kept GetClient but turned it into a wrapper that just fetches the relevant ctx and passes it to a new function NewSpannerClient.  This means that New*Client can have consistent signatures with their downstream counterparts.

**Comment by aseering** [comment]
I made updates based on your comments.  Would you mind taking another quick look to see if I got it right?

**Comment by agasheesh** [comment]
Thanks for all the changes. LGTM.

**Comment by aseering** [comment]
Tha ks!  I don't have permission to merge this PR; what's the right next step to get it merged?

**Comment by agasheesh** [comment]
@aseering I have sent you an invite for write access to this repo. You should be able to merge this PR after accepting that invite.

**Pull request merged**
